test/images/agnhost/liveness/server.go

```diff
@@ -47,12 +83,45 @@ func main(cmd *cobra.Command, args []string) {
 	})
 	http.HandleFunc("/healthz", func(w http.ResponseWriter, r *http.Request) {
 		duration := time.Since(started)
-		if duration.Seconds() > 10 {
-			w.WriteHeader(500)
-			w.Write([]byte(fmt.Sprintf("error: %v", duration.Seconds())))
-		} else {
+		if duration <= time.Duration(healthspan)*time.Second {
 			w.WriteHeader(200)
 			w.Write([]byte("ok"))
+			return
```

Comment:
if you return here then you can only test the other options later, but in theory you should return a 500.

You may add different handlers for different options to avoid to conflate all these features, leving the original handler as it is for backward compatible, and adding new handlers.

Per example, a request to `/sleep-body?time=5` makes the the server sleeps after sending a body, and does not need a flag.

same for /sleep-headers and the other flags

Reply:
Thank you for the comment. Adding handlers looks better for clarifying what is tested. I will update.